src/integrators/wrapper_integrator.cpp

```diff
@@ -81,4 +81,30 @@ void MixedBilinearToNonlinearFormIntegrator::AssembleElementGrad(const mfem::Fin
   A_->AssembleElementMatrix2(trial_el, el, Tr, elmat);
 }
 
+TransformedNonlinearFormIntegrator::TransformedNonlinearFormIntegrator(
+    std::shared_ptr<mfem::NonlinearFormIntegrator>                               R,
+    std::function<TransformedNonlinearFormIntegrator::transformed_func>          transformed,
+    std::function<TransformedNonlinearFormIntegrator::transformed_gradient_func> transformed_grad)
+    : R_(R), transformed_function_(transformed), transformed_function_grad_(transformed_grad)
+{
+}
+
+void TransformedNonlinearFormIntegrator::AssembleElementVector(const mfem::FiniteElement&   el,
+                                                               mfem::ElementTransformation& Tr,
+                                                               const mfem::Vector& elfun, mfem::Vector& elvect)
+{
+  auto transformed = transformed_function_(el, Tr, elfun);
+  R_->AssembleElementVector(el, Tr, *transformed, elvect);
+}
+
+void TransformedNonlinearFormIntegrator::AssembleElementGrad(const mfem::FiniteElement&   el,
+                                                             mfem::ElementTransformation& Tr, const mfem::Vector& elfun,
+                                                             mfem::DenseMatrix& elmat)
+{
+  auto transformed = transformed_function_(el, Tr, elfun);
+  R_->AssembleElementGrad(el, Tr, *transformed, elmat);
+  auto dense_grad_transformed = transformed_function_grad_(el, Tr, elmat);
+  elmat                       = *dense_grad_transformed;
```

Comment:
Eliminating the shared_ptr + dereference here will also allow for a return value optimization here if/when MFEM chooses to adopt move semantics.